File: packages/native_pdf_renderer/lib/src/web/resources/document_repository.dart
import 'package:native_pdf_renderer/src/web/document/document.dart';
import 'package:native_pdf_renderer/src/web/pdfjs.dart';
import 'package:native_pdf_renderer/src/web/resources/repository.dart';
import 'package:uuid/uuid.dart';

final Uuid uuid = Uuid();

class DocumentRepository extends Repository<Document> {
  /// Register document in repository
  Document register(PdfJsDoc renderer) {
    final document = Document(
      id: uuid.v1(),
      document: renderer,
    );
    set(document.id, document);
    return document;
  }

  @override
  void close(String? id) {
    get(id)!.close();
    super.close(id);
  }
}

File: packages/native_pdf_renderer/android/src/main/kotlin/io/scer/native_pdf_renderer/resources/DocumentRepository.kt
package io.scer.native_pdf_renderer.resources

import android.annotation.TargetApi
import android.graphics.pdf.PdfRenderer
import android.os.Build
import android.os.ParcelFileDescriptor
import io.scer.native_pdf_renderer.document.Document
import io.scer.native_pdf_renderer.utils.randomID

@TargetApi(Build.VERSION_CODES.LOLLIPOP)
class DocumentRepository : Repository<Document>() {
    /**
     * Register document in repository
     * @returns document id
     */
    fun register(getPair: Pair<ParcelFileDescriptor, PdfRenderer>?): Document {
        val id = randomID
        val (fileDescriptor, renderer) = getPair!!
        val document = Document(id, renderer, fileDescriptor)
        set(id, document)
        return document
    }

    public override fun close(id: String) {
        get(id).close()
        super.close(id)
    }
}